android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java

```diff
@@ -1270,6 +1271,14 @@ protected void showHelpBubble(Context context, float fx, float fy) {
       }
     }
 
+    // We can't properly display the help bubble if this token is null.  Abort and try again
+    // next time.
+    if (getWindowToken() == null) {
+      helpBubbleWindow = null;
+      ShouldShowHelpBubble = true;
```

Comment:
Sorry to be a hassle. This function is kinda messy, isn't it?

I'm thinking we should probably clear both of these at the start of this function:

```
dismissHelpBubble();
helpBubbleWindow = null;
ShouldShowHelpBubble = false;
```

Then that specific if statement would just set `ShouldShowHelpBubble = true;` and return.

Because I think some of the other code paths could leave a dangling help bubble, e.g. if you remove a keyboard or disable the bubble after it is already visible. Edge cases but it's cleaner.

Reply:
Good catch on the `dismissHelpBubble()` part!